reactor-core/src/test/java/reactor/core/publisher/LambdaMonoSubscriberTest.java

```diff
@@ -165,18 +166,27 @@ public void onNextConsumerExceptionBubblesUpDoesntTriggerCancellation() {
 		TestSubscription testSubscription = new TestSubscription();
 		tested.onSubscribe(testSubscription);
 
-		//as Mono is single-value, it cancels early on onNext. this leads to an exception
-		//during onNext to be bubbled up as a BubbledException, not propagated through onNext
-		try {
-			tested.onNext("foo");
-			fail("Expected a bubbling Exception");
-		} catch (RuntimeException e) {
-			assertThat(e).matches(Exceptions::isBubbling, "Expected a bubbling Exception")
-			             .hasCauseInstanceOf(IllegalArgumentException.class);
-		}
+		//the error is expected to be propagated through doError
+		tested.onNext("foo");
 
-		assertThat(errorHolder.get()).as("onError").isNull();
-		assertThat(testSubscription.isCancelled).as("subscription isCancelled").isFalse();
+		assertThat(errorHolder.get()).as("onError").isInstanceOf(IllegalArgumentException.class);
+		assertThat(testSubscription.isCancelled).as("subscription isCancelled").isTrue();
+	}
+
+	@Test
+	public void onNextConsumerBubblesUpErrorCallbackNotImplemented() {
+		LambdaMonoSubscriber<String> tested = new LambdaMonoSubscriber<>(
+				value -> { throw new IllegalArgumentException(); },
+				null,
+				() -> {},
+				null);
+
+		TestSubscription testSubscription = new TestSubscription();
+		tested.onSubscribe(testSubscription);
+
+		//the error is expected to be thrown as there is no error handler
+		assertThatIllegalArgumentException().isThrownBy(() -> tested.onNext("foo"));
```

Comment:
eh sorry for the back and forth. now the test fails because obviously no error handler == `ErrorCallbackNotImplemented`. So the assertion should A) check that the exception `satisfies` the `Exceptions.isErrorCallbackNotImplemented` `Predicate` and B) that its cause is the `IllegalArgumentException`.

the starting point of the assertion should thus be the more generic `assertThatExceptionOfType(RuntimeException.class).isThrownBy(() -> ...`.

Reply:
My bad, I didnt' know how I missed it when re-running the test 🙇‍♀️